dspace-api/src/main/java/org/dspace/subscriptions/ContentGenerator.java

```diff
@@ -56,8 +56,16 @@ public void notifyForSubscriptions(Context context, EPerson ePerson,
                 Locale supportedLocale = I18nUtil.getEPersonLocale(ePerson);
                 Email email = Email.getEmail(I18nUtil.getEmailFilename(supportedLocale, "subscriptions_content"));
                 email.addRecipient(ePerson.getEmail());
-                email.addArgument(generateBodyMail(context, indexableComm));
-                email.addArgument(generateBodyMail(context, indexableColl));
+
+                String bodyCommunitites = generateBodyMail(context, indexableComm);
+                String bodyCollections = generateBodyMail(context, indexableColl);
+                if (bodyCommunitites.equals(EMPTY) && bodyCollections.equals(EMPTY)) {
+                    log.debug("subscription(s) of eperson {} do(es) not match any new items: nothing to send" +
+                            " - exit silently", ePerson::getID);
+                    return;
+                }
+                email.addArgument(bodyCommunitites);
+                email.addArgument(bodyCollections);
```

Comment:
This code appears to be run even if one of `bodyCommunities` or `bodyCollections` is empty... so, it is still possible to get an empty section, as long as one section has content.